distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/Manager.java

```diff
@@ -86,6 +99,49 @@ public void start() throws Exception {
 
         leaderLatch.start();
         workersAliveChildrenCache.start();
+        startWorkPoolCleaningTask();
+    }
+
+    private void startWorkPoolCleaningTask() {
+            workPoolCleaningReschedulableScheduler.schedule(
+                    DynamicProperty.delegated(() -> Schedule.withDelay(workPoolCleanPeriodMs.get())),
+                    () -> {
+                        try {
+                            TransactionalClient.tryCommit(
+                                    curatorFramework,
+                                    CLEAN_WORK_POOL_RETRIES_COUNT,
+                                    transaction -> {
+
+                                        String workPoolVersion = paths.availableWorkPoolVersion();
+                                        int version = curatorFramework.checkExists().forPath(workPoolVersion).getVersion();
+                                        transaction.checkPathWithVersion(workPoolVersion, version);
+                                        transaction.setData(workPoolVersion, new byte[]{});
+
+                                        cleanWorkPool(transaction);
+                                    });
+                        } catch (Exception e) {
+                            log.debug("Failed to clean work-pool", e);
+                        }
+                    }
+            );
+    }
+
+    private void cleanWorkPool(TransactionalClient transaction) throws Exception {
```

Comment:
Two actors: Manager and Worker operate on subtree called WorkPool  
Suggestion: Create class AvailableWorkpoolSubTree() with methods
- pruneOutdatedJobsFromWorkPool(txClient...)
- checkAndUpdateWorkPoolVersion(txClient..)
- updateWorkPollWithWorkerAvailableJobs()

or something like that: the idea is to extract WorkPool subtree methods in separate class.

Reply:
I think the idea of creating AvailableWorkpoolSubTree is good. Maybe we should create class for every subtree in the same way? but not in this pull-request